.github/workflows/ef-tests.yml

```diff
@@ -0,0 +1,118 @@
+name: EF Tests
+
+on:
+  pull_request:
+    branches:
+      - main
+
+permissions: read-all
+
+jobs:
+  ef-tests:
+    runs-on: ubuntu-latest
+    steps:
+      # Install packages
+      - name: install packages
+        run: sudo apt-get install curl
+
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+
+      # Build Kakarot
+      - uses: actions/checkout@v3
+        with:
+          path: kakarot
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.9.13
+      - name: Load cached Poetry installation
+        id: cached-poetry
+        uses: actions/cache@v3
+        with:
+          path: ~/.local
+          key: poetry-${{ runner.os }}
+      - name: Install Poetry
+        if: steps.cached-poetry.outputs.cache-hit != 'true'
+        uses: snok/install-poetry@v1
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+          installer-parallel: true
+      - run: poetry config installer.modern-installation false
+      - name: Load cached venv
+        id: cached-poetry-dependencies
+        uses: actions/cache@v3
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ hashFiles('**/poetry.lock') }}
+      - name: Install dependencies
+        # TODO: investigate why cache is not working and remove this
+        # if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+        run: make setup
+        working-directory: kakarot
+      - name: Build
+        run: |
+          STARKNET_NETWORK=katana poetry run python ./scripts/compile_kakarot.py && make build-sol
+        working-directory: kakarot
+
+      # Make Katana dump for the build
+      - name: Checkout kakarot-rpc
+        uses: actions/checkout@v3
+        with:
+          repository: kkrt-labs/kakarot-rpc
+          path: kakarot-rpc
+      - name: Setup Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+      - name: Rust cache
+        uses: Swatinem/rust-cache@v2
+        with:
+          cache-on-failure: "true"
+      - name: make environment
+        run: ./scripts/make_with_env.sh
+        working-directory: kakarot-rpc
+      - name: pull kakarot submodule
```

Comment:
why do you pull the submodule if you build from kakarot directly? also, using this you'll have the latest kakarot@main, and not the one built just before

Reply:
The reason is that the dump-katana binary will look for two file-paths:
- kakarot-rpc/lib/kakarot
- kakarot-rpc/lib/kakarot/tests/integration/solidity_contracts

I am pulling the submodule just to get the directory structure, and then I am copying the builds and not building it { hence I am still using the latest builds! }

<img width="1236" alt="image" src="https://github.com/kkrt-labs/kakarot/assets/41180869/25850ce8-c201-4ea9-a0b2-8c390895980d">

If you want, then I can create these directories manually as well!